symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/DeprecationLogger.java

```diff
@@ -0,0 +1,28 @@
+package com.symphony.bdk.core.util;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.Objects;
+
+@Slf4j
+@API(status = API.Status.INTERNAL)
+/**
+ * Class to log deprecation message except if system property bdk.warning.mode is set to none.
+ */
+public class DeprecationLogger {
+
+  private static final boolean shouldLogDeprecationMessages = !Objects.equals(System.getProperty("bdk.warning.mode"),
+      "none");
+
+  /**
+   * Logs a WARN message except if system property bdk.warning.mode=none
+   *
+   * @param message deprecation message to log.
+   */
+  public static void logDeprecation(String message) {
```

Comment:
We should have a unit test for that (ok the code is pretty straightforward but by making it an habit helps for other parts)